browser/src/Editor/NeovimEditor/NeovimEditor.tsx

```diff
@@ -707,17 +707,21 @@ export class NeovimEditor extends Editor implements IEditor {
         .then(ids => this._actions.setCurrentBuffers(ids))
     }
 
-    private _onConfigChanged(newValues: Partial<IConfigurationValues>): void {
+    private async _onConfigChanged(newValues: Partial<IConfigurationValues>): Promise<void> {
         const fontFamily = this._configuration.getValue("editor.fontFamily")
         const fontSize = addDefaultUnitIfNeeded(this._configuration.getValue("editor.fontSize"))
         const linePadding = this._configuration.getValue("editor.linePadding")
 
         this._actions.setFont(fontFamily, fontSize)
         this._neovimInstance.setFont(fontFamily, fontSize, linePadding)
 
-        Object.keys(newValues).forEach((key) => {
+        Object.keys(newValues).forEach(async (key) => {
             const value = newValues[key]
             this._actions.setConfigValue(key, value)
+            if (key && key.includes("ui.colorscheme")) {
+                await this._themeManager.setTheme(value)
```

Comment:
Thanks for the change, @Akin909 ! 🎉 

I believe we might be setting the theme already here, so this line might not be needed:
https://github.com/onivim/oni/blob/d60fa91374b4d3f30730d96d6b635cbf56c3c3ed/browser/src/Services/Themes/index.ts#L12

The `ThemeManager` actually has an event `onThemeChanged` - maybe we could listen to that either in `Shell.tsx` or in `browser/src/index.tsx`? The reason I'm thinking about those places as opposed to `NeovimEditor` is that, eventually, we might have more than one way `NeovimEditor` instance - like for #545 . So in that case, this might get called multiple times. There's only ever one 'Shell' though at the moment.

I was actually trying to remove all the `Shell` references from `NeovimEditor`, but the title one is stuck there at the moment. But I'm hoping to get those out to fully support that 'multiplexing' scenario 😄 

Let me know what you think!

Reply:
@bryphe hopefully my changes are ok? I didn't use the listener you mentioned as looking at the activate function it seemed to be subscribed to config changes to colorsheme already just not updating the store with this info so I added an action there though I might have entirely missed the point of your suggestion 😟